crates/oxc_linter/src/fixer.rs

```diff
@@ -92,7 +91,25 @@ impl<'a> Fixer<'a> {
         output.push_str(&source_text[offset..]);
 
         let mut messages = self.messages.into_iter().filter(|m| !m.fixed).collect::<Vec<_>>();
-        messages.sort_by_key(|m| m.span);
+        messages.sort_by(|a, b| {
```

Comment:
Can you take a look at `sort_by_cached_key` and `sort_unstable_by_key`

Reply:
But our sorting is determined by two key: `offset()` and `len()`, can we use one of the above methods?

Reply:
Oh we can use tuple here